### ISSUE TITLE
FiD Report

### ISSUE THREAD

**Issue by jiuyixu25** [hustvl/LightningDiT#77]
Hi,

Thanks for your excellent work. I met an issue when I tried to reproduce LightningDiT-B/1. I got the FiD = 7.30 when training with 70k steps and FiD = 4.44 when training with 120k steps (50k generated images and other default training settings in the provided config file). However, these results are much better than those provided in the paper (e.g., Table 2). I am not sure if this is normal performance or if some other reasons lead to this. Also, which is better from your experience, LightningDiT-B/1 or LightiningDiT-B/2?

In addition, when I tried to train LightningDiT-B/1, I met a slight loss explosion:

<img width="1490" height="270" alt="Image" src="https://github.com/user-attachments/assets/45a8f428-bc29-49a4-a64d-5b820348ca7b" />

<img width="1372" height="1480" alt="Image" src="https://github.com/user-attachments/assets/3720ddb1-17cb-43ba-90c7-26d2e60bacba" />

There is no Nah value, but it is significantly larger (~2.0) than the converged loss value (~0.31). Did you observe the same thing before?

My final question is: Could you please clarify the relationship between steps and epochs? How many steps does 80 epochs equal to, 80k or 100k? Thanks for your great work again!

Best,
Joey

**Comment by alienflm** [comment]
> Hi,
> 
> Thanks for your excellent work. I met an issue when I tried to reproduce LightningDiT-B/1. I got the FiD = 7.30 when training with 70k steps and FiD = 4.44 when training with 120k steps (50k generated images and other default training settings in the provided config file). However, these results are much better than those provided in the paper (e.g., Table 2). I am not sure if this is normal performance or if some other reasons lead to this. Also, which is better from your experience, LightningDiT-B/1 or LightiningDiT-B/2?
> 
> In addition, when I tried to train LightningDiT-B/1, I met a slight loss explosion:
> 
> <img alt="Image" width="1490" height="270" src="https://private-user-images.githubusercontent.com/203024655/470579937-45a8f428-bc29-49a4-a64d-5b820348ca7b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTM0MjgzODYsIm5iZiI6MTc1MzQyODA4NiwicGF0aCI6Ii8yMDMwMjQ2NTUvNDcwNTc5OTM3LTQ1YThmNDI4LWJjMjktNDlhNC1hNjRkLTViODIwMzQ4Y2E3Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcyNVQwNzIxMjZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xZmM1ZWUxNDNhYTEwNDU2NjhlMGZlMzI1ZGYyZGRhM2YwZWFjOTQxZTk0N2Y4MGE2ZDI3MTQ5ZWMxZTA5N2NjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.9Nu-k28-JbscPobFeEca_HAfZ0BSDEgwa-ufnmScdfw"> <img alt="Image" width="1372" height="1480" src="https://private-user-images.githubusercontent.com/203024655/470581497-3720ddb1-17cb-43ba-90c7-26d2e60bacba.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTM0MjgzODYsIm5iZiI6MTc1MzQyODA4NiwicGF0aCI6Ii8yMDMwMjQ2NTUvNDcwNTgxNDk3LTM3MjBkZGIxLTE3Y2ItNDNiYS05MGM3LTI2ZDJlNjBiYWNiYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcyNVQwNzIxMjZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00YjJhMzNmYjAyNjkxOTMxZWQ4MjQyMWQ1MjgyNTZmZjczYzk4OTlhMGI4NGYzNTc4ZjM3YmJiZDQ3MzAwMGM0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.9HH9A885c5UE1g3D6VFRuh2mJD8oyBg39uLhiDSGx2s">
> There is no Nah value, but it is significantly larger (~2.0) than the converged loss value (~0.31). Did you observe the same thing before?
> 
> My final question is: Could you please clarify the relationship between steps and epochs? How many steps does 80 epochs equal to, 80k or 100k? Thanks for your great work again!
> 
> Best, Joey

I have encountered the same issue when training DiT-L/1. The loss value increases significantly (around ~2.0) which is much higher than the expected converged loss (~0.2), although there are no NaN values.

Have you found any solution or workaround for this?

I would appreciate any suggestions or insights from the community.

**Comment by JingfengYao** [comment]
Hi.

Most likely because: In order to conduct a large number of tests quickly, most of the results reported in the paper are based on FID-10k. The performance of patch=1 is significantly better than that of patch=2.

For the loss spike issue, please refer to [here](https://github.com/hustvl/LightningDiT/issues/23).

**Comment by iangellove** [comment]
@jiuyixu25 hi,thanks for share your training experience.we tried to reproduce LightningDiT-B/1,we got the FiD >= 30 when training with 80k steps.how can I achieve results like yours（70K steps fid=7）.
